Framework/API/src/SpectrumInfo.cpp

```diff
@@ -101,6 +101,27 @@ Kernel::V3D SpectrumInfo::position(const size_t index) const {
   return newPos / static_cast<double>(dets.size());
 }
 
+/// Returns the eFixed value of the spectrum with given index.
+double SpectrumInfo::eFixed(const size_t index) const {
+  double eFixed(0);
+  size_t validEFixedSize(0);
+  const auto &dets = getDetectorVector(index);
+
+  for (const auto &det : dets) {
+    const auto &detIndex = m_detectorInfo.indexOf(det->getID());
+    m_detectorInfo.setCachedDetector(detIndex, det);
+    double detectorEFixed = m_detectorInfo.eFixed(detIndex);
+
+    if (detectorEFixed != 0.0) {
```

Comment:
Though I'd rather see the use of an enum (see comment above) over a plethora of newly defined exception types.